common/table.yaml

```diff
@@ -137,3 +137,32 @@ groups:
   - data_type_inc: VectorData
     doc: Vector columns, including index columns, of this dynamic table.
     quantity: '*'
+
```

Comment:
What does this provide that isn't already provided by a regular DynamicTable?

Reply:
This class was originally created as part of the ndx-icephys-meta extension. This PR is part of a collection of PRs to integrate the ndx-icephys-meta extension with NWB.  I'd prefer to not to heavily modify the design at this point. From a user-perspective what AlignedDynamicTable provides is a table where the columns are grouped into sub-categories. In ndx-icephys-meta we have a table for storing intracellular recordings where each recording consists of a stimulus, response, and electrode and we need to be to associate additional columns with each of these categories as well as with the overall recording. AlignedDynamicTable does this by providing a main table (because it inherits from DynamicTable) and along with that a separate set of DynamicTables, one for each category, where all of these table have the same number of rows. We had decided on this approach because it makes it easy to reuse the functionality from DynamicTable and customize the behaviors (e.g., getitem) accordingly while at the same time not overloading DynamicTable itself with too much additional functionality. The matching PR for implementation of the corresponding Container class is here  https://github.com/hdmf-dev/hdmf/pull/551 and the implementation of the class which shows the added functionality is here https://github.com/hdmf-dev/hdmf/blob/b8686563f765e3193048d096ba830bfb2098501d/src/hdmf/common/table.py#L1272-L1502

Reply:
Gotcha. Is HDMF-common the right place for this then? This seems like very specific functionality, and I'm wondering if PyNWB is the right place for it. Are there any other places within PyNWB even where this would be reused?

Reply:
My thinking was that:
1. having support for categories (i.e., sub-headings) on a table is a general-purpose feature that will ultimately be useful in a number of areas, essentially any time you have metadata across different categories and in particular when you have similar fields (e.g., name) across several categories where you otherwise have to mangle names (e.g., add a prefix) to avoid name-collisions between columns.
2. there is nothing NWB-specific about this type but it really just adds some new features to DynamicTable and since DynamicTable is in hdmf-common I felt this was the right place. 
 
Another area where we need this right now is in the table for stimulus functions, because we have different types of parameters and for each parameter type we have a set of values we need to store (e.g., name and value). To avoid collisions between columns we can here currently either use compound columns or use prefix on column names, both of which are not ideal solutions because with the compound columns we can't use the ExternalResources to annotate a particular field and mangling the names of columns makes it inconvenient to interact a group columns that belong together. More broadly, I think this sis omething that would also be handy for things like the ElectrodesTable and Trials.

For NWB it doesn't make a big difference whether this is in hdmf-common or nwb-schema (i.e., if need to move it we can), but I thought having it here would make sense.